public/New-DbaAvailabilityGroup.ps1

```diff
@@ -366,6 +381,14 @@ function New-DbaAvailabilityGroup {
             return
         }
 
+        if ($IsContained -and $server.VersionMajor -lt 16) {
+           Stop-Function -Level Warning -Message "Contained availability groups are only supported in SQL Server 2022 and above"
+        }
+
+        if ($ReuseSystemDatabases -and $IsContained -eq $false) {
+           Write-Message -Level Warning -Message "Reuse system databases is only applicable in contained availability groups"
```

Comment:
I would handle this the same way like the other test: `Stop-Function` and `return`.